reference/src/layout/unions.md

```diff
@@ -6,17 +6,109 @@ not to change until an RFC ratifies them.
 
 [#13]: https://github.com/rust-rfcs/unsafe-code-guidelines/issues/13
 
-The only degree of freedom the compiler has when computing the layout of a union
-like
+### Layout of individual union fields
+
+When laying out an `union`, the compiler has to decide how the fields of the
+union are arranged. Union fields are laid out by the compiler "on top" of each
+other, that is, the bytes of one field might overlap with the bytes of another
+field - as opposed to, e.g., `struct`s, where the fields are laid out "next to"
+each other, such that the bytes of one field never overlap with the bytes of
+another field. This can be visualized as follows:
 
 ```rust,ignore
-union U { f1: T1, f2: T2 }
+[ P P [field0_ty] P P P P ]
+[ P P P P [field1_ty] P P ]
+[ P P P [field2_ty] P P P ]
```

Comment:
We'll have to come to an agreement whether to use `P P P P P` or `<------->`. I prefer the arrows, they look like the "spacers" that come up when doing visual layout of components in a UI.

Reply:
Are the inner `[]` required ?

I think it would be simpler to just treat these are:

```rust
[ P P field0_ty P P P P ]
[ P P P P field1_ty P P ]
[ P P P field2_ty P P P ]
```

> I prefer the arrows, they look like the "spacers" that come up when doing visual layout of components in a UI.

That makes sense. The advantage I see about using `Pad` is that then the `[ ]` span a list of space separated types, and for each its layout is computed following the usual rules.

Reply:
If we were to use `()` and commas as separators, the rows would look like tuples:

```rust
( P, P, field0_ty, P, P, P, P )
( P, P, P, P, field1_ty, P, P )
( P, P, P, field2_ty, P, P, P )
```

I don't know if this makes it easier or harder to understand these figures. Either way we should have good captions / glossary for them.

Reply:
> Are the inner [] required ?

I think it looks odd without them. The field should appear as a "block" somehow.

> If we were to use () and commas as separators, the rows would look like tuples:

With this it looks like every element of the tuple is equally sized. 
Really I want a picture of some wooden blocks with springs between them. I don't think commas make any sense there.

Reply:
Changed this to use <--->